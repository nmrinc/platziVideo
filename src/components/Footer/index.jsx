import React from 'react';

const Footer = () => (
  <footer className="footer">
    <a href="#">Terms</a>
    <a href="#">Privacy</a>
    <a href="#">Help Centre</a>
  </footer>
)

export default Footer;