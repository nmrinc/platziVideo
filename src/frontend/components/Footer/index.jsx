import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

const Footer = (props) => {

  const { history } = props;

  const [loc, setloc] = useState(false);

  const getLocation = (location) => {
    if (location.pathname === '/login' || location.pathname === '/signup') {
      setloc(true);
    } else {
      setloc(false);
    }
  };

  useEffect(() => {
    let didCancel = false;
    !didCancel && getLocation(props.location);
    return () => { didCancel = true; };
  // eslint-disable-next-line react/destructuring-assignment
  }, [props.location]);

  history.listen((location) => getLocation(location));

  return (
    <footer className={`footer ${loc && 'fixed--bottom'}`}>
      <button type='button'>Terms</button>
      <button type='button'>Privacy</button>
      <button type='button'>Help Centre</button>
    </footer>
  );
};

export default withRouter(connect(null, null)(Footer));
