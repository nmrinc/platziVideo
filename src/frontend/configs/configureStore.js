import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import reducer from '../reducers';

const ConfigureStore = () => {

  //@o To consume the state pre loaded from the server side, we access it by the next sentence.
  const preloadedState = window.__PRELOADED_STATE__;

  const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

  const store = createStore(
    reducer,
    //@o Pass the preloadedState as the initialState.
    preloadedState,
    composeEnhancers(applyMiddleware(thunk, logger)),
  );
  return store;
};

export default ConfigureStore;
