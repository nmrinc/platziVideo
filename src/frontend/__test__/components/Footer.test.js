/**
 * @concept Test for footer component
 * @context
 * If you want create a test for a component, you have to recreate the same structure of the component.
*/

//@a As the component uses React and Redux, your have to recreate the structure so import what's necessary.
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
//@a Import Mount from Enzyme so it help mount the component for test.
import { mount } from 'enzyme';
//@a Import the component to be tested.
import Footer from '../../components/Footer';
//@a The component is connected to a store. So import it and create its constant.
import configureStore from '../../configs/configureStore';

const store = configureStore();

//@o If you will make a series of test, you need to create a suite.
//@a  With the describe command. Pass the description and the tests.
describe('<Footer />', () => {
  //@a Create a const with the mounted component wrapped into the redux way
  const footer = mount(
    <Provider store={store}>
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    </Provider>
  );

  //@a Test if component exists
  test('Render Footer Component', () => {
    //@o Search for a node of the component
    expect(footer.length).toEqual(1);
  });

  //@a Test if 3 buttons exists into the component
  test('Footer have 3 anchors', () => {
    //@o Find the <a></a> tag into the component and test if the length is equal to 3
    expect(footer.find('button')).toHaveLength(3);
  });
});
