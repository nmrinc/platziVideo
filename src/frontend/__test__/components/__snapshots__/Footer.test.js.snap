// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Footer Snapshot 1`] = `
<footer
  className="footer false"
>
  <button
    type="button"
  >
    Terms
  </button>
  <button
    type="button"
  >
    Privacy
  </button>
  <button
    type="button"
  >
    Help Centre
  </button>
</footer>
`;
